definitions/output/all/backfill_requests.js

```diff
@@ -0,0 +1,194 @@
+    ) {
+      return "video";
+    } else if (mime_typ.includes("wasm") || ext === "wasm") {
+      return "wasm";
+    } else if (mime_typ.includes("html") || ["html", "htm"].includes(ext)) {
+      return "html";
+    } else if (mime_typ.includes("text")) {
+      // Put "text" last because it is often misused, so extension should take precedence.
+      return "text";
+    } else {
+      return "other";
+    }
+  } catch (e) {
+    return "other"; // Return "other" if there's any error
+  }
+""";
+
+CREATE TEMP FUNCTION parse_headers(headers STRING)
+RETURNS ARRAY<STRUCT<name STRING, value STRING>>
+LANGUAGE js
+AS """
+  try {
+    return JSON.parse(headers).map(header => {
+      return { name: header.name, value: header.value };
+    });
+  } catch (e) {
+    return [];
+  }
+""";
+
+INSERT INTO \`all_dev.requests_stable\` --${ctx.resolve('all', 'requests')}
+SELECT
+  DATE('${iteration.date}') AS date,
+  '${iteration.client}' AS client,
+  requests.page AS page,
+  TRUE AS is_root_page,
+  requests.page AS root_page,
+  crux.rank AS rank,
+  requests.url AS url,
+  IF(
+    SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._request_type') AS STRING) = "Document" AND
+      MIN(SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64)) OVER (PARTITION BY requests.page) = SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64),
+    TRUE,
+    FALSE
+  ) AS is_main_document,
+  get_type(JSON_VALUE(requests.payload, '$.response.content.mimeType'), get_ext_from_url(requests.url)) AS type,
+  SAFE_CAST(JSON_EXTRACT_SCALAR(payload, '$._index') AS INT64) AS index,
+  SAFE.PARSE_JSON(requests.payload, wide_number_mode => 'round') AS payload,
+  TO_JSON( STRUCT(
+    SAFE_CAST(JSON_VALUE(requests.payload, '$.time') AS INTEGER) AS time,
+    JSON_VALUE(requests.payload, '$._method') AS method,
+    NULL AS redirectUrl,
+    JSON_VALUE(requests.payload, '$.request.httpVersion') AS reqHttpVersion,
+    JSON_VALUE(requests.payload, '$.request.headersSize') AS reqHeadersSize,
+    JSON_VALUE(requests.payload, '$.request.bodySize') AS reqBodySize,
+    NULL AS reqCookieLen,
+    JSON_VALUE(requests.payload, '$.response.status') AS status,
+    JSON_VALUE(requests.payload, '$.response.httpVersion') AS respHttpVersion,
+    JSON_VALUE(requests.payload, '$.response.headersSize') AS respHeadersSize,
+    JSON_VALUE(requests.payload, '$.response.bodySize') AS respBodySize,
+    JSON_VALUE(requests.payload, '$.response.content.size') AS respSize,
+    NULL AS respCookieLen,
+    NULL AS expAge,
+    JSON_VALUE(requests.payload, '$.response.content.mimeType') AS mimeType,
+    JSON_VALUE(requests.payload, '$._cdn_provider') AS _cdn_provide,
+    JSON_VALUE(requests.payload, '$._gzip_save') AS _gzip_save,
+    NULL AS ext,
+    NULL AS format
+  )) AS summary,
+  parse_headers(JSON_QUERY(payload, '$.request.headers')) AS request_headers,
+  parse_headers(JSON_QUERY(payload, '$.response.headers')) AS response_headers,
+  response_bodies.body AS response_body
```

Comment:
```suggestion
  IF(requests.type = 'image', NULL, response_bodies.body) AS response_body
```

@tunetheweb do you think this will solve https://github.com/HTTPArchive/httparchive.org/issues/592 ?

Reply:
Well it won’t add the missing ones. But should at least remove the extra rows.

Reply:
Should we maybe check for other types too across the years, just to be sure?